src/trace-plugin-interface.js

```diff
@@ -0,0 +1,242 @@
+ * @param {function(ChildSpan)} fn A function that will be called exactly
+ * once, with a ChildSpan object exposing an interface operating on the child
+ * span.
+ * @returns The return value of calling fn.
+ */
+Transaction.prototype.runInChildSpan = function(options, fn) {
+  var that = this;
+  options = options || {};
+  var childContext = that.agent.startSpan(options.name, {},
+    options.skipFrames ? options.skipFrames + 1 : 1);
+  // If the options object passed in has the setHeader field set,
+  // use it to set trace metadata in an outgoing request.
+  if (typeof(options.setHeader) === 'function') {
+    var outgoingTraceContext = that.agent.generateTraceContext(childContext, true);
+    options.setHeader('x-cloud-trace-context', outgoingTraceContext);
+  }
+  return fn(new ChildSpan(that.agent, childContext));
+};
+
+/**
+ * Plugin constructor. Don't call directly - a Plugin object will be passed to
+ * plugins themselves
+ * TODO(kjin): Should be called something else
+ */
+function Plugin(agent) {
+  this.agent = agent;
+  this.logger = agent.logger;
+  this.config = {
+    enhancedDatabaseReporting: agent.config_.enhancedDatabaseReporting,
+    databaseResultReportingSize: agent.config_.databaseResultReportingSize
+  };
+}
+
+/**
+ * Creates a new Transaction object corresponding to an incoming request, which
+ * exposes methods operating on the root span.
+ * @param {object} options An object that specifies options for how the root
+ * span is created and propogated.
+ * @param {string} options.name The name to apply to the root span.
+ * @param {?string} options.url A URL associated with the root span, if
+ * applicable.
+ * @param {?function(string)} options.getHeader A function describing how to
+ * obtain a header field with the given field name from the incoming request
+ * assoicated with this root span. If supplied, it will be called to obtain the
+ * header field 'x-cloud-trace-context'.
+ * @param {?function(string, string)} options.setHeader A function describing
+ * how to set a header field with the given field name to a given value,
+ * respectively. If supplied, it will be called to set the header field in an
+ * outgoing request associated with this root span to the field name
+ * 'x-cloud-trace-context'.
+ * @param {?number} options.skipFrames The number of stack frames to skip when
+ * collecting call stack information for the root span, starting from the top;
+ * this should be set to avoid including frames in the plugin. Defaults to 0.
+ * @returns A new Transaction object, or null if the trace agent's policy has
+ * disabled tracing for the given set of options.
+ */
+Plugin.prototype.createTransaction = function(options) {
+  options = options || {};
+  var that = this;
+  // If the options object passed in has the getHeader field set,
+  // try to retrieve the header field containing incoming trace metadata.
+  var incomingTraceContext;
+  if (typeof(options.getHeader) === 'function') {
```

Comment:
Since you're doing this comparison a fair amount I would suggest using `is`

ex. `is.fn(options.getHeader)`